### ISSUE TITLE
Tried to set up jar file using ./gradlew build but get a string of errors for JUnit

### ISSUE THREAD

**Issue by RiyaMehta2211** [nus-cs2103-AY2324S1/forum#108]
Using Intellij 2022.3.1 Community version on MacOS Catalina version 10.15.6

When I tried to create a duke.jar file using ./gradlew build in terminal, gave me a string of errors for JUnit terms used in the tests. I have checked my program and the JUnit tests pass perfectly. 

Even under dependencies for my build.gradle, I have checked to see if JUnit is included in this
`dependencies {
    testImplementation(group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0')
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'
}`
Would really appreciate the help. Thank you.
I have provided a screenshot of the output I get which is errors on terms like @Test and assertEquals
<img width="807" alt="image" src="https://github.com/nus-cs2103-AY2324S1/forum/assets/77782419/c150561a-6942-4979-bf54-3279b8b79bb4">

**Comment by larrywang0701** [comment]
did you import those Test annoation and junit classes into your code?

**Comment by RiyaMehta2211** [comment]
Yup, I did. I could test Duke.java and these JUnit tests perfectly without gradlew using
`import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;`

**Comment by damithc** [comment]
@RiyaMehta2211 perhaps push your code to your fork and share the link here, so that others can try to reproduce the error?

BTW, does the `./gradlew test` command work?

**Comment by alvinlim277** [comment]
I encountered a similar issue, but even my static imports were not being recognised. Not entirely sure how safe this is, but I followed the top comment [here](https://stackoverflow.com/questions/13157815/why-does-intellij-idea-suddenly-not-recognize-tests-in-test-folder-anymore) and Invalidated Cache which magically solved it.

**Comment by RiyaMehta2211** [comment]
> @RiyaMehta2211 perhaps push your code to your fork and share the link here, so that others can try to reproduce the error?
> 
> BTW, does the `./gradlew test` command work?

Hi prof the ./gradlew test gives me a “Build Failed” with the same “error: cannot find symbol”  I have pushed the commit to my master branch for easier reference

**Comment by RiyaMehta2211** [comment]
> I encountered a similar issue, but even my static imports were not being recognised. Not entirely sure how safe this is, but I followed the top comment [here](https://stackoverflow.com/questions/13157815/why-does-intellij-idea-suddenly-not-recognize-tests-in-test-folder-anymore) and Invalidated Cache which magically solved it.

Yup I see, will try that

**Comment by RiyaMehta2211** [comment]
> I encountered a similar issue, but even my static imports were not being recognised. Not entirely sure how safe this is, but I followed the top comment [here](https://stackoverflow.com/questions/13157815/why-does-intellij-idea-suddenly-not-recognize-tests-in-test-folder-anymore) and Invalidated Cache which magically solved it.

Also did u happen to just invalidate and restart or clear file cache and history as well? I tried just invalidating and restarting but that didn’t work

**Comment by RiyaMehta2211** [comment]
Also are there any corrections to be made to my Project Structure settings?

<img width="807" alt="Screenshot 2023-09-05 at 6 48 39 PM" src="https://github.com/nus-cs2103-AY2324S1/forum/assets/77782419/3bc448f7-72a8-4668-bc71-13d086bc9a0e">
<img width="807" alt="Screenshot 2023-09-05 at 6 49 35 PM" src="https://github.com/nus-cs2103-AY2324S1/forum/assets/77782419/2ed60c48-6c7d-45b7-a883-a29e76e4d09e">
Thank you!

**Comment by alvinlim277** [comment]
> > I encountered a similar issue, but even my static imports were not being recognised. Not entirely sure how safe this is, but I followed the top comment [here](https://stackoverflow.com/questions/13157815/why-does-intellij-idea-suddenly-not-recognize-tests-in-test-folder-anymore) and Invalidated Cache which magically solved it.
> 
> Also did u happen to just invalidate and restart or clear file cache and history as well? I tried just invalidating and restarting but that didn’t work

I only invalidated cache, didn't do anything else. Our issues might be slightly different then...

**Comment by junhonglow** [comment]
You will need to create a test folder as seen [here](https://se-education.org/guides/tutorials/junit.html), additionally you may need to revert the changes you made to the dependencies portion of your gradle.build file as well, I had to undo the modifications for it to build.

**Comment by RiyaMehta2211** [comment]
Yup, I created a test folder and transferred my test files into the folder as per the guidelines but the issue still persists.

**Comment by damithc** [comment]
> Yup, I created a test folder and transferred my test files into the folder as per the guidelines but the issue still persists.

@RiyaMehta2211 Push this latest code to the fork and share the link here. Then, someone can just clone it and troubleshoot it locally.

**Comment by RiyaMehta2211** [comment]
Yup prof, I have pushed the code and am sharing the link below:
[https://github.com/RiyaMehta2211/ip/commit/82636ff30406894479c69ec46f746a4a14e9858b](url)
This code is part of the commit for JUnit tests with all the test files being in the test folder

**Comment by RiyaMehta2211** [comment]
I have manually defined the assertEquals method to get rid of some issues, now the only issue is @Test is not being recognized. Is there any way to change this as @Test is required, and I also tried modifying dependencies to no avail

**Comment by damithc** [comment]
@RiyaMehta2211 is [this](https://github.com/RiyaMehta2211/ip) your latest code? I still can't see the test folder.
The commit before has a `test` folder but not in the correct place. Gradle doesn't work if the files are in the expected locations.

As @junhonglow mentioned, have another look at https://se-education.org/guides/tutorials/junit.html (see 'Conventions to follow' section)

**Comment by RiyaMehta2211** [comment]
Hi prof, this is my latest commit in the url below, was having trouble in committing due to conflicts in the branches
[https://github.com/RiyaMehta2211/ip/commit/b784562b2217368427600852f61ff0c0debc6073](url). I have also tried putting the test folder both inside and outside the duke package with the test files inside, but gradle is unable to build. I managed to make a jar file, but only after removing the @Test notations which seem to be the only issue at this point in time.

**Comment by damithc** [comment]
@RiyaMehta2211 these two are very different:

Actual, as per https://github.com/RiyaMehta2211/ip/commit/b784562b2217368427600852f61ff0c0debc6073
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/1673303/4d52b432-9eb2-4bc6-8a87-c028675d8ea2)


Required:
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/1673303/4bdd71d1-baaf-4c9e-8781-ad21f28c85b3)

**Comment by RiyaMehta2211** [comment]
Hi I understand now,
And did the exact parallel path with the relevant files and directories in the test folder, at the same level as the main folder.

After these changes, I tried invalidating cache and rebuilding after this process: But the following error is yielded - 
<img width="1313" alt="Screenshot 2023-09-06 at 12 44 10 AM" src="https://github.com/nus-cs2103-AY2324S1/forum/assets/77782419/80bc28c2-020d-471e-9040-cfe9d4efa2f5">
May I ask if I am doing anything wrong here?
I have attached the link for my latest commit for reference which shows the updated location of the JUnit tests in the test folder: [https://github.com/RiyaMehta2211/ip/commit/c6b86b2d863afc5d521a92c5a64962d857e679c5](url)
To be specific, I get Task :compileTestJava FAILED. 

Thank you all so much for the help!
